### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by zilm13**
## PR Description

The following dependencies have later milestone versions:
> - com.diffplug.spotless:com.diffplug.spotless.gradle.plugin [6.15.0 -> 6.16.0]
> - com.github.ben-manes.versions:com.github.ben-manes.versions.gradle.plugin [0.45.0 -> 0.46.0]

Updated

> - com.launchdarkly:okhttp-eventsource [3.0.0 -> 4.1.0]
     https://github.com/launchdarkly/okhttp-eventsource

Breaking changes, created separate issue #6890 
> - com.squareup.okhttp3:mockwebserver [4.9.3 -> 5.0.0-alpha.11]
     https://square.github.io/okhttp/
> - com.squareup.okhttp3:okhttp [4.9.3 -> 5.0.0-alpha.11]
     https://square.github.io/okhttp/

5.x all alphas, using latest stable instead

> - io.github.classgraph:classgraph [4.8.154 -> 4.8.155]
     https://github.com/classgraph/classgraph

Updated     
> - io.prometheus:simpleclient [0.9.0 -> 0.16.0]
     http://github.com/prometheus/client_java

Matching Besu's metrics-core
> - io.vertx:vertx-core [4.3.8 -> 4.4.0]
> - io.vertx:vertx-web [4.3.8 -> 4.4.0]

Updated
> - org.apache.logging.log4j:log4j-api [2.19.0 -> 2.20.0]
     https://logging.apache.org/log4j/2.x/
> - org.apache.logging.log4j:log4j-core [2.19.0 -> 2.20.0]
     https://logging.apache.org/log4j/2.x/
> - org.apache.logging.log4j:log4j-slf4j-impl [2.19.0 -> 2.20.0]
     https://logging.apache.org/log4j/2.x/
> - org.apache.logging.log4j:log4j-slf4j2-impl [2.19.0 -> 2.20.0]
     https://logging.apache.org/log4j/2.x/

 Updated    
>  - org.jetbrains.kotlin:kotlin-stdlib [1.8.10 -> 1.8.20-Beta]
     https://kotlinlang.org/

 We have latest stable    
> - org.web3j:core [4.9.6 -> 5.0.0]
     https://web3j.io

 We have latest stable, 5.0.0 old error version
> - org.webjars:swagger-ui [4.15.5 -> 4.16.0-alpha.1]
     http://webjars.org

 We have latest stable

> Failed to determine the latest version for the following dependencies (use --info for details):
> - org.hyperledger.besu:besu-datatypes
> - org.hyperledger.besu:evm
> - org.hyperledger.besu:plugin-api
     23.1.1-RC1
> - org.hyperledger.besu.internal:config
> - org.hyperledger.besu.internal:core
> - org.hyperledger.besu.internal:metrics-core
     23.1.1-RC1

  Updated to 22.10.4, 23.1.0 looks like the latest stable but breaks on Java version compatibility. Errors like this:
  ```
  No matching variant of org.hyperledger.besu.internal:core:23.1.0 was found. The consumer was configured to find a runtime of a library compatible with Java 11, packaged as a jar, preferably optimized for standard JVMs, and its dependencies declared externally but:
          - Variant 'apiElements' capability org.hyperledger.besu.internal:core:23.1.0 declares a library, packaged as a jar, and its dependencies declared externally:
              - Incompatible because this component declares an API of a component compatible with Java 17 and the consumer needed a runtime of a component compatible with Java 11
  ```            

> Gradle release-candidate updates:
> - Gradle: [7.6 -> 8.0.1]

Release candidate, updated to the latest stable 7.6.1

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionSha256Sum=7ba68c54029790ab444b39d7e293d3236b2632631fb5f2e012bb28b4ff669e4b
```
Comment (StefanBratanov) [inline gradle/wrapper/gradle-wrapper.properties:3]: We can get the sha256Sum from here: https://gradle.org/release-checksums/ think it's used in the validate gradle wrapper stage

**Comment by lucassaldanha** [comment]
Mostly ok. Just need to get the checksum for gradle wrapper fixed.

**Comment by zilm13** [comment]
@StefanBratanov @lucassaldanha 
Actually it was tricky to make it check the checksum, so I added instructions to our internal documentation

**Review by StefanBratanov — APPROVED** [review]
LGTM

**Pull request merged**